### PR TITLE
fix: doc url link type

### PR DESCRIPTION
**Pull request by rainy-me**
fix: #12033

I did some debugging and found the cause looks like to be some doc links' `LinkType` are kept as `Shortcut` which don't make sense for url links.
This PR should resolve both problems in the origin issue, but aside this PR, more work are needed for doc_links.

about `LinkType`: https://github.com/raphlinus/pulldown-cmark/blob/f29bd1e228913690e5092c9594e4e607423ff0aa/src/lib.rs#L191-L210

### CODE REVIEW

File: crates/hir-def/src/attr.rs [hunk 1]
```diff
@@ -853,6 +853,23 @@ impl<'attr> AttrQuery<'attr> {
             .iter()
             .filter(move |attr| attr.path.as_ident().map_or(false, |s| s.to_smol_str() == key))
     }
+
+    pub fn find_string_value_in_tt(self, key: &'attr str) -> Option<&SmolStr> {
+        if !self.exists() {
+            return None;
+        }
```
Comment (Veykril) [inline crates/hir-def/src/attr.rs:860]: This check is actually unnecessary

File: crates/hir-def/src/attr.rs [hunk 2]
```diff
@@ -853,6 +853,23 @@ impl<'attr> AttrQuery<'attr> {
             .iter()
             .filter(move |attr| attr.path.as_ident().map_or(false, |s| s.to_smol_str() == key))
     }
+
+    pub fn find_string_value_in_tt(self, key: &'attr str) -> Option<&SmolStr> {
```
Comment (Veykril) [inline crates/hir-def/src/attr.rs:857]: Let's add a comment showing an example for what this specifically does (as the name is not too descriptive, but I also do not have a better one). Something like

/// #[doc(html_root_url = "url")] 
///       ^^^^^^^^^^^^^ key

should suffice

**Review by Veykril — COMMENTED** [review]
@bors delegate

**Comment by lnicola** [comment]
changelog fix fix keyword doc links in hover

image

**Pull request merged**